beacon-chain/sync/pending_attestations_queue.go

```diff
@@ -391,13 +414,19 @@ func (s *Service) savePending(root [32]byte, pending any, isEqual func(other any
 	s.blkRootToPendingAtts[root] = append(s.blkRootToPendingAtts[root], pending)
 }
 
-func pendingAggregatesAreEqual(a, b ethpb.SignedAggregateAttAndProof) bool {
+// pendingAggregatesAreEqual checks if two pending aggregate attestations are equal.
+// If filterOnAggregatorIndex is false, two aggregates only differ by their aggregator index will be considered equal.
+func pendingAggregatesAreEqual(a, b ethpb.SignedAggregateAttAndProof, filterOnAggregatorIndex bool) bool {
```

Comment:
for the record I don't mind boolean controllers, but even some linters would recommend here a named wrapper that has the extra validation. I think it's fine as is.